### PR TITLE
[Matrix] change to new C++ PVR interface way

### PR DESCRIPTION
**Pull request by AlwinEsch**
Changes related to xbmc/xbmc#16485.

This make the addon to the new C++ addon interface.
With missing backend not runtime tested!

About them I looked to hold his code style and hope stil correct. 

Also are them set to handle one instance way. If more as one backend can be possible for them, it need the `CAddonBase` and `CInstancePVRClient` separated (this currently not supported in Kodi, but addon interface prepared for it). 
If backend only one time possible then this the better way for me.

Hope my changes are mostly OK for you  :smile:

### CODE REVIEW

**Comment by aassif** [comment]
Impressive!
Couldn't have done any better!
Thank you!

File: src/Freebox.cpp [hunk 1]
```diff
@@ -2111,75 +2205,57 @@ PVR_ERROR Freebox::DeleteTimer (const PVR_TIMER & timer, bool force)
...
   {
-    // Heading.
-    PVR_FREEBOX_STRING_CHANNEL_SOURCE,
-    // Entries.
-    PVR_FREEBOX_STRING_CHANNEL_SOURCE_AUTO,
-    PVR_FREEBOX_STRING_CHANNEL_SOURCE_IPTV,
-    PVR_FREEBOX_STRING_CHANNEL_SOURCE_DVB
+    kodi::GetLocalizedString (PVR_FREEBOX_STRING_CHANNEL_SOURCE_AUTO),
+    kodi::GetLocalizedString (PVR_FREEBOX_STRING_CHANNEL_SOURCE_IPTV),
+    kodi::GetLocalizedString (PVR_FREEBOX_STRING_CHANNEL_SOURCE_DVB)
```
Comment (AlwinEsch) [inline src/Freebox.cpp:2211]: About here need to confirmed that works, as it defined in `static` I'm not sure with the calls inside.

Comment (aassif) [reply src/Freebox.cpp:2211]: I haven't built Kodi since I moved to Ubuntu 20.04.
Might take quite some time ;)

Comment (aassif) [reply src/Freebox.cpp:2211]: I successfully built the addon yesterday and it works.
Except it keeps the first language when switching to another one.
I think we should just drop the `static` attribute.

Comment (AlwinEsch) [reply src/Freebox.cpp:2211]: Thanks a lot for test!
Yes then makes a static no sense and remove it there, thought not on begin about that, where then only one time called and no more correct if language changed.

Comment (aassif) [reply src/Freebox.cpp:2211]: You're welcome!
Will you make the change yourself or should I make a PR on your forked repo?
It's the first time I need to make a change before merging :)

Comment (AlwinEsch) [reply src/Freebox.cpp:2211]: I make change about, no problem with git :smile:

**Comment by AlwinEsch** [comment]
Performed update of request, the request on Kodi itself is finished and only need now all addons a bit recheck and small changes.
For here:
- "CheckSettings..." changed to "GetSettings..." where now use a default value.
- ADDON_GetStatus is removed, as it obsolete and no more called
 - Have added a commit to automate the addon version define, so a hand change no more needed (where I overseen always before :smile:)

**Comment by aassif** [comment]
I recently discovered `GetAddonInfo` which I was about to use but the `#define` is way better!
Thanks!

**Comment by aassif** [comment]
Ready to merge.

**Comment by AlwinEsch** [comment]
The request in Kodi is in, after the Ubuntu package build of them is ready can it in here too (tomorrow).

Why comes here no AppVeyor, Travis and Jenkins build?

**Comment by aassif** [comment]
> Why comes here no AppVeyor, Travis and Jenkins build?

What do you mean?

**Comment by AlwinEsch** [comment]
Has not seen any test builds here e.g. like here https://github.com/flubshi/pvr.waipu/pull/76 (the "All checks have passed")

**Edit:** Also not sure if then after release a from Jenkins created binary becomes publicated.

**Pull request merged**